### PR TITLE
bootstrap hooks

### PR DESCRIPTION
**Pull request by akalman**
This will give very basic support for hooks during bootstrap.

I went with something as simple as possible since I have basically no data revolving around use cases.  There's a lot that's missing here, hooks based on user, having hooks load from dependent ruby/puppet modules, only running under certain conditions if we really want to create a "puppet-like" experience.  But this will get the job done.

I went with a directory over a naming convention because I thought it'd be harder to get wrong and puppet already has semantic directory structures.  Although I suppose it also has semantic names as well since boxen loads files with specific names by default.

### CODE REVIEW

**Comment by akalman** [comment]
actually wait a bit for this.  In a fit of delirium I forgot to test to this to make sure it works.

**Comment by jacobbednarz** [comment]
Thanks for this @akalman! Let me know when you're ready for a review and I'll take a look.

**Comment by akalman** [comment]
ok consider this haphazardly tested!

**Comment by akalman** [comment]
@jacobbednarz any update on this?

**Comment by jacobbednarz** [comment]
I've been going back and forth with myself on this for the last week trying to weigh up the pros and cons because this has the potential to slow down Boxen runs if we're not careful about it.

File: script/bootstrap [hunk 1]
```diff
@@ -60,6 +60,15 @@ set -e
         /usr/bin/gem install bundler -v '~> 1.10.5' --no-rdoc --no-ri
 }
 PATH="/Library/Ruby/bin:$(gem which bundler | sed -e 's!lib/bundler.rb!bin!'):$PATH"
+
+# Run any user defined hooks before calculating checksum
+# so if the gemfile gets modified we can re-bundle if so.
+for hook in script/bootstrap-hooks/*; do
+    if [ -x $hook ] && [ ! -d $hook ]; then
+        $hook
```
Comment (jacobbednarz) [inline script/bootstrap]: Maybe add some feedback in here on what script is being executed?

**Comment by jacobbednarz** [comment]
@akalman is this still something you're interested in landing?

**Pull request closed without merging**